If showing Setting->Updates, the gateway is restarting

After update gateway to 0.53 a have problem show page updates.
Updates page not show any text.
![gateway - updates](https://user-images.githubusercontent.com/43097974/45804571-c0d72980-bcbc-11e8-8d05-33c06930e885.png)

syslog:

`Sep 20 09:49:59 gateway run-app.sh[9131]: SyntaxError: Unexpected end of JSON input`
`Sep 20 09:49:59 gateway run-app.sh[9131]:     at JSON.parse (<anonymous>)`
`Sep 20 09:49:59 gateway run-app.sh[9131]:     at /home/pi/mozilla-iot/gateway/build/webpack:/src/controllers/updates_controller.js:19:1`
`Sep 20 09:49:59 gateway run-app.sh[9131]:     at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:511:3)`
`Sep 20 09:49:59 gateway run-app.sh[9131]: npm ERR! code ELIFECYCLE`
`Sep 20 09:49:59 gateway run-app.sh[9131]: npm ERR! errno 1`
`Sep 20 09:49:59 gateway run-app.sh[9131]: npm ERR! things-gateway@0.5.3 start: 'webpack && node build/gateway.js "--check-wifi"'`
`Sep 20 09:49:59 gateway run-app.sh[9131]: npm ERR! Exit status 1`
`Sep 20 09:49:59 gateway run-app.sh[9131]: npm ERR!`
`Sep 20 09:49:59 gateway run-app.sh[9131]: npm ERR! Failed at the things-gateway@0.5.3 start script.`
`Sep 20 09:49:59 gateway run-app.sh[9131]: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.`
`Sep 20 09:49:59 gateway run-app.sh[9131]: npm ERR! A complete log of this run can be found in:`
`Sep 20 09:49:59 gateway run-app.sh[9131]: npm ERR!     /home/pi/.npm/_logs/2018-09-20T07_49_59_715Z-debug.log`
`Sep 20 09:50:09 gateway systemd[1]: mozilla-iot-gateway.service: Service hold-off time over, scheduling restart.`
`Sep 20 09:50:09 gateway systemd[1]: Stopped Mozilla IoT Gateway Client.`
`Sep 20 09:50:09 gateway systemd[1]: Started Mozilla IoT Gateway Client.`

file 2018-09-20T07_49_59_715Z-debug.log
`0 info it worked if it ends with ok`
`1 verbose cli [ '/home/pi/.nvm/versions/node/v8.12.0/bin/node',`
`1 verbose cli   '/home/pi/.nvm/versions/node/v8.12.0/bin/npm',`
`1 verbose cli   'start',`
`1 verbose cli   '--',`
`1 verbose cli   '--check-wifi' ]`
`2 info using npm@6.4.1`
`3 info using node@v8.12.0`
`4 verbose run-script [ 'prestart', 'start', 'poststart' ]`
`5 info lifecycle things-gateway@0.5.3~prestart: things-gateway@0.5.3`
`6 info lifecycle things-gateway@0.5.3~start: things-gateway@0.5.3`
`7 verbose lifecycle things-gateway@0.5.3~start: unsafe-perm in lifecycle true`
`8 verbose lifecycle things-gateway@0.5.3~start: PATH: /home/pi/.nvm/versions/node/v8.12.0/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/pi/mozilla-iot/gateway/node_modules/.bin:/home/pi/.nvm/versions/node/v8.12.0/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin`
`9 verbose lifecycle things-gateway@0.5.3~start: CWD: /home/pi/mozilla-iot/gateway`
`10 silly lifecycle things-gateway@0.5.3~start: Args: [ '-c', 'webpack && node build/gateway.js "--check-wifi"' ]`
`11 silly lifecycle things-gateway@0.5.3~start: Returned: code: 1  signal: null`
`12 info lifecycle things-gateway@0.5.3~start: Failed to exec start script`
`13 verbose stack Error: things-gateway@0.5.3 start: 'webpack && node build/gateway.js "--check-wifi"'`
`13 verbose stack Exit status 1`
`13 verbose stack     at EventEmitter.<anonymous> (/home/pi/.nvm/versions/node/v8.12.0/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:301:16)`
`13 verbose stack     at emitTwo (events.js:126:13)`
`13 verbose stack     at EventEmitter.emit (events.js:214:7)`
`13 verbose stack     at ChildProcess.<anonymous> (/home/pi/.nvm/versions/node/v8.12.0/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)`
`13 verbose stack     at emitTwo (events.js:126:13)`
`13 verbose stack     at ChildProcess.emit (events.js:214:7)`
`13 verbose stack     at maybeClose (internal/child_process.js:915:16)`
`13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)`
`14 verbose pkgid things-gateway@0.5.3`
`15 verbose cwd /home/pi/mozilla-iot/gateway`
`16 verbose Linux 4.14.52-v7+`
`17 verbose argv "/home/pi/.nvm/versions/node/v8.12.0/bin/node" "/home/pi/.nvm/versions/node/v8.12.0/bin/npm" "start" "--" "--check-wifi"`
`18 verbose node v8.12.0`
`19 verbose npm  v6.4.1`
`20 error code ELIFECYCLE`
`21 error errno 1`
`22 error things-gateway@0.5.3 start: 'webpack && node build/gateway.js "--check-wifi"'`
`22 error Exit status 1`
`23 error Failed at the things-gateway@0.5.3 start script.`
`23 error This is probably not a problem with npm. There is likely additional logging output above.`
`24 verbose exit [ 1, true ]`

--- a/src/controllers/updates_controller.js
+++ b/src/controllers/updates_controller.js
@@ -16,12 +16,19 @@ function readVersion(packagePath) {
         reject(err);
         return;
       }
-      const pkg = JSON.parse(data);
-      if (!semver.valid(pkg.version)) {
-        reject(new Error(`invalid gateway semver: ${pkg.version}`));
-        return;
+
+      try {
+        const pkgJson = JSON.parse(data);
+
+        if (!semver.valid(pkgJson.version)) {
+          reject(new Error(`Invalid gateway semver: ${pkgJson.version}`));
+          return;
+        }
+
+        resolve(pkgJson.version);
+      } catch (e) {
+        reject(e);
       }
-      resolve(pkg.version);
     });
   });
 }
@@ -105,19 +112,26 @@ UpdatesController.get('/status', async function(request, response) {
   // if (gateway_failed.version > thisversion) {
   //  update failed, last attempt was ctime of gateway_failed
   // }
-  const currentVersion = await readVersion('package.json');
+  const currentVersion = pkg.version;
+
   const oldStats = await stat('../gateway_old/package.json');
   let oldVersion = null;
+  if (oldStats) {
+    try {
+      oldVersion = await readVersion('../gateway_old/package.json');
+    } catch (e) {
+      console.error('Failed to read ../gateway_old/package.json:', e);
+    }
+  }
 
   const failedStats = await stat('../gateway_failed/package.json');
   let failedVersion = null;
-
-  if (oldStats) {
-    oldVersion = await readVersion('../gateway_old/package.json');
-  }
-
   if (failedStats) {
-    failedVersion = await readVersion('../gateway_failed/package.json');
+    try {
+      failedVersion = await readVersion('../gateway_failed/package.json');
+    } catch (e) {
+      console.error('Failed to read ../gateway_failed/package.json:', e);
+    }
   }
 
   if (failedVersion && semver.gt(failedVersion, currentVersion)) {

